pkg/kvstore/etcd.go

```diff
@@ -701,6 +708,9 @@ func (e *etcdClient) Update(ctx context.Context, key string, value []byte, lease
 	if lease {
 		duration := spanstat.Start()
 		_, err := e.client.Put(ctx, key, string(value), client.WithLease(e.GetLeaseID()))
+		if err == v3rpcErrors.ErrLeaseNotFound {
```

Comment:
Can you move this into a function? Please also add a comment to explain why we need to handle this error in this special way. If possible with a reference to upstream details.